README.md

```diff
@@ -659,10 +659,290 @@ To begin coding the basic execution functionality of our VM, we only need these
+			}
+		case ResMul:
+			if dst != nil && op1 != nil {
+				dst_felt, dst_is_felt := dst.GetFelt()
+				op1_felt, op1_is_felt := op1.GetFelt()
+				if dst_is_felt && op1_is_felt && !op1_felt.IsZero() {
+					return memory.NewMaybeRelocatableFelt(dst_felt.Div(op1_felt)), dst, nil
+
+				}
+			}
+		}
+	}
+	return nil, nil, nil
+}
+```
+
+##### DeduceOp1
+
+The method deduces the value of `Op1` if possible (based on `dst` and `Op0`) it also deduces `res` if possible.
+If the instruction opcode is `AssertEq` a switch case is used to check what the `ResLogic` is, if it is a `ResOp1` then the value of op1 is equal to the dst operand. If `ResLogic` is `ResAdd` op1 is deduced from the substraction of `dst` with `op0`. Finally if it is `ResMul` `op1` is deduced from the division of `dst` with `op0`, in all the cases `res` is equal to `dst`. if none of the former cases apply then nil is returned.
+
+```go
+func (vm *VirtualMachine) DeduceOp1(instruction Instruction, dst *memory.MaybeRelocatable, op0 *memory.MaybeRelocatable) (*memory.MaybeRelocatable, *memory.MaybeRelocatable, error) {
+	if instruction.Opcode == AssertEq {
+		switch instruction.ResLogic {
+		case ResOp1:
+			return dst, dst, nil
+		case ResAdd:
+			if op0 != nil && dst != nil {
+				dst_rel, err := dst.Sub(*op0)
+				if err != nil {
+					return nil, nil, err
+				}
+				return &dst_rel, dst, nil
+			}
+		case ResMul:
+			dst_felt, dst_is_felt := dst.GetFelt()
+			op0_felt, op0_is_felt := op0.GetFelt()
+			if dst_is_felt && op0_is_felt && !op0_felt.IsZero() {
+				res := memory.NewMaybeRelocatableFelt(dst_felt.Div(op0_felt))
+				return res, dst, nil
+			}
+		}
+	}
+	return nil, nil, nil
+}
+
 ```
+##### ComputeRes
+
+If the Res value has not been deduced in the previous steps then it is computed based on the `Op0` and `Op1` values and the res logic stored in the instruction.
+If `ResLogic` is `ResOp1` then `res` is equal to `op1`. If it is `ResAdd` then `res` is deduced from the addition of `op0` with `op1`. If it is `ResMul` `res` is deduced from the multiplication of `op0` and `op1`. Otherwise `res` is nil.
 
+```go
+func (vm *VirtualMachine) ComputeRes(instruction Instruction, op0 memory.MaybeRelocatable, op1 memory.MaybeRelocatable) (*memory.MaybeRelocatable, error) {
+	switch instruction.ResLogic {
+	case ResOp1:
+		return &op1, nil
+
+	case ResAdd:
+		maybe_rel, err := op0.Add(op1)
+		if err != nil {
+			return nil, err
+		}
+		return &maybe_rel, nil
+
+	case ResMul:
+		num_op0, m_type := op0.GetFelt()
+		num_op1, other_type := op1.GetFelt()
+		if m_type && other_type {
+			result := memory.NewMaybeRelocatableFelt(num_op0.Mul(num_op1))
+			return result, nil
+		} else {
+			return nil, errors.New("ComputeResRelocatableMul")
+		}
+
+	case ResUnconstrained:
+		return nil, nil
+	}
+	return nil, nil
+}
+```
+
+##### DeduceDst
+
+if the destination value has not been calculated before then it is deduced based on the Res value. If the opcode is an `AssertEq` then dst is equal res.
```

Comment:
Is `Res value` the value of the `res` operand? If it is, please rename `Res value` to `res operand`.

Reply:
Also,
```suggestion
If the destination value has not been calculated before then it is deduced based on the Res value. If the opcode is an `AssertEq` then dst is equal res.
```